File: irrgarten/Monster.java
package irrgarten;

public class Monster extends LabyrinthCharacter{
    private final static int INITIAL_HEALTH = 5;
    private final String name;
    private final float intelligence;
    private final float strength;
    private float health;
    private int row;
    private int col;

    /*public Monster(String name, float intelligence, float strength) {
        this.name = name;
        this.intelligence = intelligence;
        this.strength = strength;
        health = INITIAL_HEALTH;
    }*/

    public Monster(String name, float intelligence, float strength, float health, String name1, float intelligence1, float strength1) {
        super(name, intelligence, strength, health);
        this.name = name1;
        this.intelligence = intelligence1;
        this.strength = strength1;
    }

    public boolean dead() {
        return this.health <= 0;
    }

    public float attack() {
        return Dice.intensity(this.strength);
    }

    public void setPos(int row, int col) {
        this.row = row;
        this.col = col;
    }

    public String toString() {
        return "Name: " + this.name + " Intelligence: " + this.intelligence + " Strentgh: " + this.strength + " Health: " + this.health;
    }

    public void gotWounded() {
        this.health--;
    }

    public boolean defend(float receivedAttack) {

        float defensiveEnergy = Dice.intensity(intelligence);
        if (defensiveEnergy < receivedAttack){
            gotWounded();
        }

        return dead();
    }
}

File: irrgarten/Monster.rb
module Irrgarten
  require_relative 'Dice'
  class Monster
    #Atributos de la clase
    @@INITIAL_HEALTH=5

    #Constructor
    def initialize(name, intelligence,strength)
      @name = name
      @intelligence = intelligence
      @strength =strength
      @health = @@INITIAL_HEALTH
      @row = -1
      @col = -1
    end
    #Dice si el monstruo esta muerto
    def dead()
      if @health !=0
        return false
      else
        return true
      end
    end
    #Intensidad del ataque del monstruo
    def attack()
      return Dice.intensity(@strength)
    end
    #Por implementar
    def defend(recievedAttack)
      isDead=self.dead
      if !isDead
        defensiveEnergy=Dice.intensity(@intelligence)
        if defensiveEnergy<recievedAttack
          self.gotWounded
          isDead=self.dead
        end
      end
      return isDead
    end
    #Setter de la posicion del monstruo
    def setPos(row,col)
      @row=row
      @col=col
    end
    #Print del estado del monstruo
    def to_s
      "El monstruo #{@name} tiene una inteligencia de #{@intelligence}, una fuerza de #{@strength}, una salud de #{@health} y se encuentra en [#{@row},#{@col}]"
    end
    #Metod que reduce la salud de un monstruo cuando es atacado.
    private
    def gotWounded()
      if @health > 0
        @health-= 1
      end
    end

  end
end